docs/proposals/images/machinepool-machines/inframachinepool-delete-replicas.plantuml

```diff
@@ -0,0 +1,33 @@
+@startuml inframachinepool-delete-replicas
+:InfraMachinePool delete replicas reconciliation;
+:fetch all InfraMachinePoolMachines;
+:fetch owner Machine for each InfraMachinePoolMachine;
+if (any owner Machine is missing) then
+  :wait for owner Machine to be created;
+  :requeue;
```

Comment:
it's a bit strange to have the requeue end in a stop, should this loop back to the first step?